src/sentry/sentry_metrics/indexer/limiters/cardinality.py

```diff
@@ -81,7 +81,14 @@ def check_cardinality_limits(
         hash_to_offset = {}
         for key, message in messages.items():
             org_id = message["org_id"]
-            if not sample_modulo("sentry-metrics.cardinality-limiter.orgs-rollout-rate", org_id):
+            if use_case_id == UseCaseKey.PERFORMANCE and not sample_modulo(
```

Comment:
You don't need to check the use_case_id for every message. You can find out the option to use based on use_case_id once outside the loop and then use the option for each message.

Reply:
good point, fixed